libcontainerd/supervisor/remote_daemon.go

```diff
@@ -144,17 +153,16 @@ func (r *remote) getContainerdPid() (int, error) {
 }
 
 func (r *remote) getContainerdConfig() (string, error) {
```

Comment:
Nit: I am not a fan of functions with side-effects masquerading as getters. And returning the path to the config file hides the fact there is a single source of truth.
```suggestion
func (r *remote) writeContainerdConfig(path string) error {
```

Reply:
heh. I agree!! "get" is a pretty bad name for this (and I had to look a few times "where is that thing created??)

I initially had a commit that renamed it to `generateContainerdConfig()`, but combined with https://github.com/moby/moby/pull/43946, that no longer made sense (that PR changes it to skip the generating and only returns the name when using a custom config), so I removed that change.

Let me look at some of that in the follow-up as well.

Reply:
"ensure" ? :eyes:

Reply:
ensure could work; it stands out enough to make anyone reading the code aware "it's doing something"